search-provider/eks-article-metadata-query-provider-dbus.xml

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE node PUBLIC
+"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+
+<!--
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General
+ Public License along with this library; if not, see <http://www.gnu.org/licenses/>.
+-->
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <interface name="com.endlessm.ArticleMetadata">
+    <method name="Query">
+      <arg type="as" name="Shards" direction="out" />
+      <arg type="aa{sv}" name="Results" direction="out" />
+      <arg type="a{sv}" name="Query" direction="in" />
+    </method>
+    <method name="Shards">
```

Comment:
When would one use this method?

Reply:
Its used in the companion-app-service side in order to get a list of shards for an application when you already know the contentId and don't need to do a query. We can't cache the shard list on the service side because it is meant to be stateless.